tests/application_development/code_relocation/linker_arm_sram2.ld

```diff
@@ -34,6 +34,11 @@ MEMORY
 #ifdef CONFIG_SRAM2
     SRAM2                  (wx) : ORIGIN = (CONFIG_SRAM_BASE_ADDRESS + RAM_SIZE2), LENGTH = RAM_SIZE2
 #endif
+
+#ifdef CONFIG_SOC_SERIES_S32K3_M7
+    IVT_HEADER              (r) : ORIGIN = (CONFIG_FLASH_BASE_ADDRESS + CONFIG_IVT_HEADER_OFFSET),
+                                  LENGTH = CONFIG_IVT_HEADER_SIZE
+#endif
```

Comment:
better to move the changes to board config, directly to another linker script for S32 series, if they are not using
#include <zephyr/arch/arm/aarch32/cortex_m/scripts/linker.ld>

Reply:
This SoC is using the arch linker file https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nxp_s32/s32k/linker.ld#L15. But this test is using a custom linker script, therefore not using the ones at SoC level.

Reply:
@manuargue , for such platform specific case, you need create a new linker script and add a new config and change the testcase.yml create a new case for S30 platform. the linker_arm_sram2.ld is based on platform that using the #include <zephyr/arch/arm/aarch32/cortex_m/scripts/linker.ld>

Reply:
S32K3 SoC is using the arch linker `#include <zephyr/arch/arm/aarch32/cortex_m/scripts/linker.ld>` already, as I pointed out in https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nxp_s32/s32k/linker.ld#L15. The issue is this test is using  `CONFIG_CUSTOM_LINKER_SCRIPT="linker_arm_sram2.ld"` which overrides the SoC linker so the needed memory region is not present, and that's why is introduced here. I could move it to a separate linker file, but I consider is still part of Arm test, and not a special case. There are many platforms that define a SoC specific linker, so I wonder on which SoCs this test is actually running.